WalletWasabi.Gui/Controls/NoparaPasswordBox.cs

```diff
@@ -328,11 +335,13 @@ private bool OnTextInput(string text, bool isPaste)
 				Sb.Append(text);
 			}
 
-			if (handledCorrectly && Sb.Length > Constants.MaxPasswordLength) // Ensure the maximum length.
+			if (handledCorrectly && Sb.Length > Constants.MaxPasswordLength) // We should not get here, ensure the maximum length.
 			{
 				Sb.Remove(Constants.MaxPasswordLength, Sb.Length - Constants.MaxPasswordLength);
 				handledCorrectly = false; // Should play beep sound not working on windows.
+				_ = DisplayWarningAsync(PasswordTooLongMessage);
```

Comment:
Any way to avoid this being called asynchronously?

Reply:
We are doing the same elsewhere so let it be, this will be replaced by notifications.